torch/csrc/distributed/rpc/python_functions.cpp

```diff
@@ -220,6 +246,13 @@ PyRRef pyRemotePythonUdf(
         userRRef->rrefId().toIValue(),
         userRRef->forkId().toIValue());
 
+    if (rf != nullptr) {
+      // attach the recordFunction object to the future, so it can be set as
+      // completed when the future finishes.
+      rf->setOverrideThreadId(true);
+      fm->attachRecordFunction(std::move(rf));
+    }
+
```

Comment:
Can we move this inside `sendMessageWithAutograd` and as result it will be done in a common place? I recall we discussed this offline in person, but I forgot if there was a good reason for not doing it inside `sendMessageWithAutograd`?

Reply:
I think we decided not to do that since we weren't sure if every RPC starting from python layer eventually calls into `sendMessageWithAutograd`. Although upon further inspection it looks like we can do this. Although a downside is we'd have to pass the shared_ptr another layer down,.